lib/irb/cmd/debug.rb

```diff
@@ -25,11 +25,16 @@ def execute(*args)
           return
         end
 
+        command = nil
+        if debug_command
+          command = ['irb', nil, debug_command]
+        end
+
         # To make debugger commands like `next` or `continue` work without asking
         # the user to quit IRB after that, we need to exit IRB first and then hit
         # a TracePoint on #debug_break.
         file, lineno = IRB::Irb.instance_method(:debug_break).source_location
-        DEBUGGER__::SESSION.add_line_breakpoint(file, lineno + 1, oneshot: true, hook_call: false)
+        DEBUGGER__::SESSION.add_line_breakpoint(file, lineno + 1, oneshot: true, hook_call: false, command: command)
```

Comment:
I'm not sure if inserting a new breakpoint with command is the best way to implement these commands. It may mess up with level computation (not sure).

I think ideally we should make `irb` an UI of debugger. That means the command can be [passed directly into debugger's session](https://github.com/ruby/debug/blob/d7f181c322beadaa456a4db7af5298526dd5b1e1/lib/debug/session.rb#L404-L409).

Reply:
> I'm not sure if inserting a new breakpoint with command is the best way to implement these commands. It may mess up with level computation (not sure).

I'm not sure what's different in the level computation between `debug` command and non-`debug` commands. My understanding is that it's primarily handled by the monkey-patched `capture_frames` on the oneshot breakpoint issued by IRB. If it's broken for non-`debug` commands, I'm not sure if it would work for `debug` either. We move to the debugger with the same implementation, and we start executing commands from the session we entered in the same way by a human input or this `command` option. 

> I think ideally we should make irb an UI of debugger. 

I agree. And this is what I meant in https://github.com/ruby/irb/issues/425#issuecomment-1318264770 to give up for the Ruby 3.2 release. My idea is that, in Ruby 3.2, it's okay to make these commands a superset / thin wrapper of `debug`, and hopefully make IRB a UI of debugger in Ruby 3.3 or so if we want.

If there's something that doesn't work with this implementation, I'll probably just give it up instead of working on the perfect UI integration. But I assume it would mean that `debug` command is also broken on that case.

Reply:
> hopefully make IRB a UI of debugger in Ruby 3.3 or so for these commands if we want.

I don't mean adding an irb UI to the debug gem. I think we can customise a UI class and then [initialise the debugger with it](https://github.com/ruby/debug/blob/d7f181c322beadaa456a4db7af5298526dd5b1e1/lib/debug/session.rb#L2116-L2123). Instead of passing a `UI_LocalConsole.new`, we pass `UI_IRBConsole.new`, for example.

Reply:
Yes, I knew about the `UI_*` classes, and we were talking about the same thing.

Reply:
I don't stop you from making that improvement if you want, but my point is that I believe the current design should work for the cases where typing the same command on the debugger mode started by the `debug` command you added works, and it saves us some time if you're concerned about the 1-month schedule. (And again, I'll give up if I find some unexpected blocker.)

Reply:
> Yes, I knew about the UI_* classes, and we were talking about the same thing

I see, sorry. I didn't know you expect it to take much more effort. What obstacles do you currently see with this approach? Just curious.

Reply:
I think the main problem for me is that I still don't have much understanding of the debug.gem implementation; I only spent a couple of nights playing with it.

I already know enough to write up this PR within a few minutes, but I'm sure it would take more than a few minutes to learn how the UI is integrated and write an original UI class for IRB. It's probably easy for you, but I assume I'm the only person contributing to this feature since you seem to want to spend more time on auto-completion and documentation. And the ratio of my OSS time that I can spend for IRB is not large enough to accomplish the other implementation. But I also believe that the quality of the current design is as good as `debug` command, and we've effectively agreed that `debug` command is a Ruby 3.2-worthy quality.

Reply:
I revised the implementation and I'd say it's working really well now. Everything is working as intended, so I don't think we need to work on the UI thing.